source/common/router/router.cc

```diff
@@ -301,12 +307,9 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
         callbacks_);
   }
 
-  // Do a common header check. We make sure that all outgoing requests have all HTTP/2 headers.
-  // These get stripped by HTTP/1 codec where applicable.
+  // Finally assert that an http scheme was selected
+  // in the before continuing.
```

Comment:
I think this assert can be moved right below the call to `FilterUtility::setUpstreamScheme`? Also, can you reflow and fix the grammar in the comment?